Start of (many) summary stats table attempts

melojih opened the pull request:
- started table code in 0_summary_table.do, which outputs summary_table.tex in output folder

melojih committed:
Resolved some minor typos and started summary table code

- started table code in 0_summary_table.do, which outputs summary_table.tex in output folder

melojih committed:
Attempted rest of summary stat table

Followed example in analysis.do very closely, but was unable to figure out how to testrun table code

amarchenko26 commented on code/tables/0_summary_table.do:
```diff
@@ -1,4 +1,63 @@
+
+* --- start of table here --- *
+
+// what does this do?
```
when you wanna drop something or change ur data without changing it permanently, preserve/restore lets you make those changes, and then restore the data to the way it way before the preserve statement 

also these things you can google

amarchenko26 commented on code/tables/output/robustness_city.tex:
```diff
@@ -316,7 +316,7 @@
                     &     (4.758)         &                     &                     &                     &                     &                     &                     \\
 [1em]
-La Ca�ada Flintridge&      -79.27\sym{***}&                     &                     &                     &                     &                     &                     \\
+La Cañada Flintridge&      -79.27\sym{***}&                     &                     &                     &                     &                     &                     \\
```
we shouldn't even have all these FEs in here, it should be keep(city coefficients) in the esttab command

amarchenko26 commented on code/tables/0_summary_table.do:
```diff
@@ -23,9 +23,12 @@ global repository `c(pwd)' //returns current working directory
 * --- start of table here --- *
 
-// what does this do?
+// Question: what does this do?
```
answered in previous commit

amarchenko26 commented on code/tables/0_summary_table.do:
```diff
@@ -35,19 +38,52 @@ file open f using "$repository/code/tables/output/summary_table.tex", write repl
-// hi pick up here
-
-
+local ncat price number_of_reviews accommodates bedrooms bathrooms beds amenities cleaning_fee extra_people minimum_nights first_review
```
are you sure first_review is an non-cat?i actually dont remember....go back to the code where i create it, it's either a date or just a part of a data.

amarchenko26 commented on code/tables/0_summary_table.do:
```diff
@@ -35,19 +38,52 @@ file open f using "$repository/code/tables/output/summary_table.tex", write repl
+local ncat price number_of_reviews accommodates bedrooms bathrooms beds amenities cleaning_fee extra_people minimum_nights first_review
+local cat property_type room_type instant_bookable cancellation_policy
+
+	foreach i in `ncat'{ //loops over noncategorical variables
```
does this need `quotes' here? i'm not sure...seems like it would but for some reason i feel like it might not? idk

melojih replied:
it does! `quotes' are necessary to point to foreach loop to the local varlist

amarchenko26 commented on code/tables/0_summary_table.do:
```diff
@@ -35,19 +38,52 @@ file open f using "$repository/code/tables/output/summary_table.tex", write repl
+			}
+			//write means
+			//write label
+			file write f " `: var label `i'' & " %4.3f (`NO_mean_`i'') " & " %4.3f (`YES_mean_`i'') " "
```
`file write f " `: var label `i'' & " %4.3f (`White_mean_`i'') " & " %4.3f (`Black_mean_`i'') " "`

amarchenko26 replied:
instead of NO_mean_i and YES_mean_i, you need the levelsof race. Because check out where that local is created in line #58 - it starts with `f', which is the level of race. so you're going to be 4? 5? locals, all starting with black_mean_nameofvariablehere, white_mean_nameofvariablehere, hispanic_mean_nameofvariablehere

amarchenko26 commented on code/tables/0_summary_table.do:
```diff
@@ -35,19 +38,52 @@ file open f using "$repository/code/tables/output/summary_table.tex", write repl
+					loc `f'_prop_`c'_`i' = `numerator'/`denominator'
+				}
+				local row_label : label `i' `c'
+				file write f " `row_label' & " %4.3f (`NO_prop_`c'_`i'') " & " %4.3f (`YES_prop_`c'_`i'') " "
```
this is wrong for the same reason

amarchenko26 reviewed (commented):
okay here you go! should be pretty good btw just need to thing about what exact locals you're getting out of the loops.

melojih committed:
resolving comments 1.30.19

- addressed Anya's comments on previous commit

melojih committed:
print debugging 1.31.19

melojih committed:
Anya, the code on line 56 is driving me crazy

please help
I've been print debugging that one line for ages
I don't know what I'm missing, any guidance would be much appreciated

melojih committed:
more useless debugging 2.1.19

- changed matsize back to 11000 (error saying matsize not large enough otherwise)
- dropped in a couple print statements in 0_summary_table.do to facilitate debugging

melojih committed:
successful debugging 2.5.19

- testing sum stat table code on non-categorical variables only
- created sample variable

melojih committed:
improved table 2.7.19

- inserted note at foot of table
- added vertical lines to separate first two columns
- added "Regression Sample" header spanning 5 columns
- added a few more variables and "Observations" row at bottom
